pkg/transformer/kubernetes/kubernetes.go

```diff
@@ -640,10 +640,15 @@ func (k *Kubernetes) Deploy(komposeObject kobject.KomposeObject, opt kobject.Con
 	fmt.Println("We are going to create Kubernetes Deployments, Services" + pvcStr + "for your Dockerized application. \n" +
 		"If you need different kind of resources, use the 'kompose convert' and 'kubectl create -f' commands instead. \n")
 
-	client, namespace, err := k.GetKubernetesClient()
+	client, ns, err := k.GetKubernetesClient()
+	namespace := opt.Namespace
+	if len(namespace) == 0 {
```

Comment:
In In order to avoid all of this `len` and checking against 0, have a look at how we check if a default value is changed in `convert.go`.

For example:
```
      IsDeploymentFlag:            cmd.Flags().Lookup("deployment").Changed,
```
on line 73.

Instead, you can check to see if the --namespace flag has been set via:

```
      IsNamespaceFlag:            cmd.Flags().Lookup("namespace").Changed,
```

and simply check with

```
if opt.IsNamespaceFlag {
...
}
```

in the resulting code.